### PR TITLE
Fixed the username text position, changed the username text color due to low visibility on sand

### PR DESCRIPTION
**Pull request by streanga-sarmis**
Changed the name color due to low visibility on sand;
Fixed the position for the username text and cached the divisions.

### CODE REVIEW

File: src/com/redomar/game/entities/Player.java [hunk 1]
```diff
@@ -38,6 +43,8 @@ public Player(LevelHandler level, int x, int y, InputHandler input,
 		this.shirtCol = shirtCol;
 		this.colour = Colours.get(-1, 111, shirtCol, faceCol);
 		fireRate = Small.FIRE_RATE;
+		// Perfectly matches the text over the head
+		nameOffset = (userName.length() / 2) * fontCharSize - ((userName.length() & 1) == 0 ? fontCharSize / 2 : 0);
```
Comment (DylanMeeus) [inline src/com/redomar/game/entities/Player.java]: Might consider splitting the equation from the ternary for readability.

File: src/com/redomar/game/entities/Player.java [hunk 2]
```diff
@@ -28,6 +28,11 @@
 	private boolean[] swimType;
 	private int[] swimColour;
 	private int fireRate = 0;
+	
+	// Need to add a class for constants
+	private int fontCharSize = 8;
```
Comment (DylanMeeus) [inline src/com/redomar/game/entities/Player.java]: Could probably be "final" :-)

**Comment by DylanMeeus** [comment]
Overall good commit (code wise), didn't yet check it in the game itself though! 
:+1:

File: src/com/redomar/game/entities/Player.java (hunk 1 above)
Comment (streanga-sarmis) [reply src/com/redomar/game/entities/Player.java]: You are right about that, didn't observe its over 80 characters long.

**Comment by redomar** [comment]
### Thoughts
I agree it is difficult to see the name tag over the sandy areas, An excellent idea.

### Issue I found
The code works only if you input a desired username, however if you blast through without typing a username it isnt aligned at all. Guest names are generated only when the username box is empty.
screen shot 2018-02-12 at 12 13 09
screen shot 2018-02-12 at 12 16 50

### My solution
However, I have a solution moving [This line](https://github.com/streanga-sarmis/JavaGame/blob/ef061fe0d6bfc50935d03895d13f821038a16b53/src/com/redomar/game/entities/Player.java#L47) into this [part](https://github.com/streanga-sarmis/JavaGame/blob/ef061fe0d6bfc50935d03895d13f821038a16b53/src/com/redomar/game/entities/Player.java#L197-L210). I'm guessing it is because it is after the 'temporary/guest' name has been made.

### What do you think
Do you know why this is happening? Do you have another way to deal with this or are you going to try my solution, I'm open to opinions.

**Comment by streanga-sarmis** [comment]
Yes, this is because the username initially is blank and gets modified on the go to be replaced with the "guest" one.
Its an easy fix, just need to change the offset when the username gets generated after it was blank, I will upload it in a sec.

**Comment by streanga-sarmis** [comment]
It looks a bit ugly (2 lines of code are repeating, could just use a function) + if the name is only 1 character long, its going to recalculate the offset again, but its perfectly aligned over the head.

**Comment by redomar** [comment]
Thanks for your time and effort

**Comment by streanga-sarmis** [comment]
No problem, the pleasure is mine.

**Comment by redomar** [comment]
I messed up sorry, would you like to push the changes again, I overwritten the master directory when I was merging from previous work

**Comment by redomar** [comment]
I've not used GitHub/Git in ages and I foolishly pushed a bad merge that was old

**Comment by streanga-sarmis** [comment]
I made another pr to be easier.

**Pull request merged**